### ISSUE TITLE
BaseHorizonError.result_xdr is populated incorrectly in __init__.py

### ISSUE THREAD

**Issue by JakeUrban** [StellarCN/py-stellar-base#236]
I noticed this bug while working on changes to the [reference anchor server](https://github.com/stellar/stellar-anchor-server), specifically when a user account has not created a trustline with the sender's asset.

The bug is that for a failed POST /transaction response, `result_xdr` lives under `extras`:
```
{ 
   "type":"https://stellar.org/horizon-errors/transaction_failed",
   "title":"Transaction Failed",
   "status":400,
   "detail":"The transaction failed when submitted to the stellar network. The `extras.result_codes` field on this response contains further details.  Descriptions of each code can be found at: https://www.stellar.org/developers/guides/concepts/list-of-operations.html",
   "extras":{ 
      "envelope_xdr":"AAAAAHjfQaW1e3VvimoPLH7Gwfe/fxOJVO71FM0EJ4jEd5IcAAAAZAAAFaAAAAALAAAAAAAAAAAAAAABAAAAAAAAAAEAAAAAsKEcCgAODRBPBrBrossSKVlxED4OkyZsqwfIz9j0efoAAAABVVNEAAAAAACuRBYV6IBZsHa28YoFkbngQmx2gI0FpYLj6K8Y1iRQKgAAAAAFXSNwAAAAAAAAAAHEd5IcAAAAQNe5jfChoeEynCoCFlIhiKZQLtoy+FoFsqMI9FS7a9NdiQO+gE+Vxvim1YEMDeCG9N4o+LGavCD7hPuNnup6bgw=",
      "result_codes":{ 
         "transaction":"tx_failed",
         "operations":[ 
            "op_src_no_trust"
         ]
      },
      "result_xdr":"AAAAAAAAAGT/////AAAAAQAAAAAAAAAB/////QAAAAA="
   }
}
```

The fix:
In `exceptions.py` under the `BaseHorizonError.__init__` function:
```
self.result_xdr: Optional[str] = message.get("result_xdr")
``` 
should be 
```
self.result_xdr: Optional[str] = message.get("extras", {}).get("result_xdr")
```

**Comment by overcat** [comment]
Thanks, it will be fixed today.

**Comment by overcat** [comment]
Fixed in **2.0.0b4**.
https://pypi.org/project/stellar-sdk/2.0.0b4/
